### DISCUSSION TITLE
Caching in Oak with Deno Deploy

### DISCUSSION THREAD

**Discussion by ghost** [oakserver/oak#489] [Q&A]
I am using Oak server with Deno Deploy to fetch a data from a database, it's easy, intuitive to use Oak. But when I am trying to cache the fetched data from a database, I can't get it working well because I am making a post request to extract only needed data. Also I can't get any caching library in deno.land website or built-in caching in Oak. I am using cache-control but it's not working well. What do you recommend me to do?

**Comment by EduM22** [comment]
You could implement a caching layer in your application. I would recommend https://www.upstash.com/

**Reply by ghost** [reply]
Yes! I can use these kinds of a service, but they tend add complexity. What I like to have is built-in caching in Oak Server. We might be able to use the service worker cache API or some kind of cache object. Now I have moved to Get Request, cache control works, but I need some kind of edge caching.
